crates/swc_ecma_transforms_compat/src/es2020/opt_chaining.rs

```diff
@@ -495,23 +502,25 @@ impl OptChaining {
                 ..
             }) => {
                 let obj_span = obj.span();
-                let is_super_access = match *obj {
+                let is_super_access = match **obj {
                     Expr::Member(MemberExpr {
                         obj: ExprOrSuper::Super(..),
                         ..
                     }) => true,
                     _ => false,
                 };
 
-                let (left, right, alt) = match *obj {
-                    Expr::Ident(..) => (obj.clone(), obj, e.expr),
-                    _ if is_simple_expr(&obj) && self.c.pure_getter => (obj.clone(), obj, e.expr),
+                let (left, right, alt) = match **obj {
+                    Expr::Ident(..) => (obj.clone(), obj.clone(), e.expr.take()),
```

Comment:
Isn't it safe to use `.take()` for second object?

Reply:
it is safe in terms of compile, and it makes obj to `Invalid` as it's taken over and made existing test fail.